src/tests/java/htsjdk/samtools/cram/CRAIIndexTest.java

```diff
@@ -77,57 +93,81 @@ private boolean allFoundEntriesIntersectQueryInFind(final List<CRAIEntry> index,
+        Assert.assertNotNull(span);
+        final long[] coordinateArray = span.toCoordinateArray();
+        Assert.assertEquals(coordinateArray.length, 2);
+        Assert.assertEquals(coordinateArray[0] >> 16, entry.containerStartOffset);
+        Assert.assertEquals(coordinateArray[1] & 0xFFFF, 1);
     }
 
     @Test
-    public void testFindLastAlignedEntry() {
+    public void testCraiFromFile() throws IOException {
         final List<CRAIEntry> index = new ArrayList<CRAIEntry>();
-        Assert.assertEquals(-1, CRAIIndex.findLastAlignedEntry(index));
-
-        // Scan all allowed combinations of 10 mapped/unmapped entries and assert the found last aligned entry:
-        final int indexSize = 10;
-        for (int lastAligned = 0; lastAligned < indexSize; lastAligned++) {
-            index.clear();
-            for (int i = 0; i < indexSize; i++) {
-                final CRAIEntry e = new CRAIEntry();
-
-                e.sequenceId = (i <= lastAligned ? 0 : -1);
-                e.alignmentStart = i;
-                index.add(e);
-            }
-            // check expectations are correct before calling findLastAlignedEntry method:
-            Assert.assertTrue(index.get(lastAligned).sequenceId != -1);
-            if (lastAligned < index.size() - 1) {
-                Assert.assertTrue(index.get(lastAligned + 1).sequenceId == -1);
-            }
-            // assert the the found value matches the expectation:
-            Assert.assertEquals(CRAIIndex.findLastAlignedEntry(index), lastAligned);
-        }
+        final CRAIEntry entry = new CRAIEntry();
+        entry.sequenceId = 0;
+        entry.alignmentStart = 1;
+        entry.alignmentSpan = 2;
+        entry.sliceOffset = 3;
+        entry.sliceSize = 4;
+        entry.containerStartOffset = 5;
+        index.add(entry);
+
+        final File file = File.createTempFile("test", ".crai");
+        file.deleteOnExit();
+        final FileOutputStream fos = new FileOutputStream(file);
+        final GZIPOutputStream gos = new GZIPOutputStream(fos);
+        CRAIIndex.writeIndex(gos, index);
+        gos.close();
+
+        final SAMSequenceDictionary dictionary = new SAMSequenceDictionary();
+        dictionary.addSequence(new SAMSequenceRecord("1", 100));
+
+        final SeekableStream baiStream = CRAIIndex.openCraiFileAsBaiStream(new SeekableBufferedStream(new SeekableFileStream(file)), dictionary);
+        Assert.assertNotNull(baiStream);
+
+        final DiskBasedBAMFileIndex bamIndex = new DiskBasedBAMFileIndex(baiStream, dictionary);
+        final BAMFileSpan span = bamIndex.getSpanOverlapping(entry.sequenceId, entry.alignmentStart, entry.alignmentStart);
+        Assert.assertNotNull(span);
+        final long[] coordinateArray = span.toCoordinateArray();
+        Assert.assertEquals(coordinateArray.length, 2);
+        Assert.assertEquals(coordinateArray[0] >> 16, entry.containerStartOffset);
+        Assert.assertEquals(coordinateArray[1] & 0xFFFF, 1);
```

Comment:
Looks like these two test cases share much of the same code -- can you extract a shared helper method?